src/test/java/org/broadinstitute/hellbender/tools/CRAMRefIntegrationTest.java

```diff
@@ -0,0 +1,62 @@
+package org.broadinstitute.hellbender.tools;
+
+import org.broadinstitute.hellbender.CommandLineProgramTest;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.read.SamAssertionUtils;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.File;
+
+public final class CRAMRefIntegrationTest extends CommandLineProgramTest{
+
+    private static final File TEST_DATA_DIR = new File("/Users/edwardk/Documents/hellbender/src/test/resources/org/broadinstitute/hellbender/tools");
+
+    @Override
+    public String getTestedClassName() {
+        return PrintReads.class.getSimpleName();
+    }
+
+
+    @Test(dataProvider="testingDataNoRef", expectedExceptions = UserException.class)
+    public void testNoRef(String fileIn, String extOut) throws Exception {
+        final File outFile = File.createTempFile(fileIn + ".", extOut);
+        outFile.deleteOnExit();
+        File ORIG_BAM = new File(TEST_DATA_DIR, fileIn);
+        final String[] args = new String[]{
+                "--input" , ORIG_BAM.getAbsolutePath(),
+                "--output", outFile.getAbsolutePath()
+        };
+        runCommandLine(args);
+    }
+
+    @Test(dataProvider="testingDataWrongRef", expectedExceptions = UserException.class)
+    public void testWrongRef(String fileIn, String extOut, String referenceFile) throws Exception {
+        final File outFile = File.createTempFile(fileIn + ".", extOut);
+        outFile.deleteOnExit();
+        File ORIG_BAM = new File(TEST_DATA_DIR, fileIn);
+        File reference = new File(TEST_DATA_DIR, referenceFile);
+        final String[] args = new String[]{
+                "--input" , ORIG_BAM.getAbsolutePath(),
+                "--output", outFile.getAbsolutePath(),
+                "-R", reference.getAbsolutePath()
+        };
+        runCommandLine(args);
+    }
+
+    @DataProvider(name="testingDataNoRef")
+    public Object[][] testingDataNoRef() {
+        return new String[][]{
+                {"split_reads.cram", ".sam"},
+        };
+    }
+
+    @DataProvider(name="testingDataWrongRef")
+    public Object[][] testingDataWrongRef() {
+        return new String[][]{
+                {"split_reads.cram", ".sam", "cramtest.fasta"},
```

Comment:
Are the sequence dictionaries compatible between `split_reads.cram` and `cramtest.fasta`? We want a test case involving **compatible** dictionaries (so that the reference mismatch won't be caught by the sequence dictionary validation).